File: api/app/interfaces/schemas/memory.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.domain.models.memory_entry import MemoryScope, MemorySource


class MemoryEntryCreateRequest(BaseModel):
    title: str
    content: str
    tags: list[str] = Field(default_factory=list)
    scope: MemoryScope = MemoryScope.GLOBAL
    session_id: str | None = None


class MemoryEntryUpdateRequest(BaseModel):
    title: str | None = None
    content: str | None = None
    tags: list[str] | None = None
    scope: MemoryScope | None = None
    session_id: str | None = None


class MemoryEntryResponse(BaseModel):
    id: str
    scope: MemoryScope
    session_id: str | None
    title: str
    content: str
    tags: list[str]
    source: MemorySource
    last_used_at: datetime | None
    use_count: int
    created_at: datetime
    updated_at: datetime


class MemoryEntryListResponse(BaseModel):
    entries: list[MemoryEntryResponse]

File: ui/src/lib/api/types/memory.ts
// ==================== 记忆管理 ====================

export type MemoryScope = "global" | "session";
type MemorySource = "manual" | "auto_extracted" | "tool_save";

export type MemoryEntry = {
  id: string;
  scope: MemoryScope;
  session_id?: string | null;
  title: string;
  content: string;
  tags: string[];
  source: MemorySource;
  last_used_at?: string | null;
  use_count: number;
  created_at?: string;
  updated_at?: string;
};

export type MemoryEntriesData = {
  entries: MemoryEntry[];
};

export type SessionMemoryData = {
  planner: Array<Record<string, unknown>>;
  react: Array<Record<string, unknown>>;
};
